src/BuildingRegistry/Building/BuildingUnitState.cs

```diff
@@ -143,12 +143,9 @@ private void When(BuildingUnitWasCorrectedFromRetiredToRealized @event)
         {
             Status = BuildingUnitStatus.Realized;
 
-            if (!string.IsNullOrWhiteSpace(@event.DerivedExtendedWkbGeometry))
-            {
-                BuildingUnitPosition = new BuildingUnitPosition(
-                    new ExtendedWkbGeometry(@event.DerivedExtendedWkbGeometry),
-                    BuildingUnitPositionGeometryMethod.DerivedFromObject);
-            }
+            BuildingUnitPosition = new BuildingUnitPosition(
```

Comment:
The DerivedExtendedWkbGeometry property on the event can be null in case the position should not be changed.